Pokedex/Main/MainTableViewController.swift

```diff
@@ -9,10 +9,10 @@
 import UIKit
 
 final class MainTableViewController: UITableViewController {
-
-    private let client = PokemonClient()
+    
+    private let viewModel = MainViewModel()
     private let cellReuseID = "PokemonCell"
-    private var allPokemons: AllPokemonsResponse? {
+    private var allPokemons: [String]? {
```

Comment:
This doesn't need to be Optional. Always try to have non optional types because they are much more easy to work with because you don't need unwrap them.

Reply:
Try to match ViewController properties names to your outputs as much as it makes sense. In that case, that would be `pokemonNames`.